editor/utils/editor-media-upload/index.js

```diff
@@ -31,18 +31,24 @@ export default function editorMediaUpload( {
 } ) {
 	const postId = select( 'core/editor' ).getCurrentPostId();
 
-	const errorHandler = ( { file, sizeAboveLimit, generalError } ) => {
+	const errorHandler = ( { file, type } ) => {
 		let errorMsg;
-		if ( sizeAboveLimit ) {
-			errorMsg = sprintf(
-				__( '%s exceeds the maximum upload size for this site.' ),
-				file.name
-			);
-		} else if ( generalError ) {
-			errorMsg = sprintf(
-				__( 'Error while uploading file %s to the media library.' ),
-				file.name
-			);
+		switch ( type ) {
+			case 'SIZE_ABOVE_LIMIT':
+				errorMsg = sprintf(
+					__( '%s exceeds the maximum upload size for this site.' ),
+					file.name
+				);
+				break;
+			case 'MIME_TYPE_NOT_ALLOWED_FOR_USER':
```

Comment:
Is using MIME_TYPE_NOT_ALLOWED instead of MIME_TYPE_NOT_ALLOWED_FOR_USER and being more generic in the error message 'Sorry, this file type is not allowed.' an option? Or there is a reason to be more specific?
The get_allowed_mime_types function says "Retrieve list of allowed mime types and file extensions." does not specifies that this restriction is user specific or related with security reasons. I think some restrictions may be user specific and others may not be user specific.

Reply:
The reason I'm saying `_FOR_USER ` is to differentiate it from the other prop we currently have, `allowedTypes`. That one is **also** doing mime type checking. However, it is being used for arbitrary restrictions that have nothing to do with WP user privileges, such as only allowing image files for the Image block.

Right now, the check for `allowedTypes` is not throwing errors (which it should). When we eventually make that enhancement, we should probably rename it so it isn't as confusing. Something that doesn't use the word "allowed", like maybe `accept` or `suggestedTypes`.

I'm using the error message "Sorry, this file type is not permitted for security reasons" for consistency, since that is the message used by Core that is returned when you upload a restricted file.

> I think some restrictions may be user specific and others may not be user specific.

This is true, but the impression I got from [the code](https://developer.wordpress.org/reference/functions/get_allowed_mime_types/) was that it is ultimately a function of the user (e.g. `get_allowed_mime_types( int|WP_User $user = null )` and `apply_filters( 'upload_mimes', $t, $user )`). Anyway, the verbose naming (_FOR_USER) is just intended for differentiation, so we could possibly get rid of it when `allowedTypes` is renamed 🙂